### PR TITLE
Implement transformation for TensorFlow 2 Map Function (aka tf.map_fn)

### PR DESCRIPTION
**Pull request by rkazants**
**Root cause analysis:** In several TF2 OD models (MobileNet SSD V2, CenterNet) we have a preprocessor block based on tf.map_fn primitive that slices multiple inputs, transform them, and concatenate multiple intermediate outputs at each iteration of While Loop. Currently, the MO tool does not support such block.

**Solution:** Extend existing transformations for TF2 Keras RNN operations that are particular case of tf.map_fn when it has just one sliced input and one concatenated output. The proposed extension detects sub-graphs (from the main-graph and body-graph) for slicing and concatenation and utilizes `axis` attribute of Loop operation for such operations.

**Ticket:** 47280

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests - N/A due to complexity of such unit-tests implementation
* [x]  Framework operation tests
* [x]  Transformation tests - N/A due to complexity of such unit-tests implementation. Now it covers by TF2 Keras RNN layer tests
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: model-optimizer/extensions/ops/loop.py [hunk 1]
```diff
@@ -61,6 +61,26 @@ def get_body_node_by_internal_id(loop_node: Node, internal_id: int):
             'Expected 0 or 1 node with `internal_layer_id`={}, {} found'.format(internal_id, len(suitable_nodes))
         return suitable_nodes[0] if len(suitable_nodes) == 1 else None
 
+    @staticmethod
+    def get_external_node_by_internal_id(loop_node: Node, internal_layer_id: int):
+        """
+
+        :param loop_node:
+        :param internal_layer_id:
+        :return:
```
Comment (rkazants) [inline model-optimizer/extensions/ops/loop.py:70]: add description for this function

**Comment by rkazants** [comment]
For some unknown reason I can't send new commits to this branch. It is probably due to my occasional attempt to push changes into upstream.
